docs/concepts/ipc.md

```diff
@@ -10,40 +10,53 @@ Most IPC within Santa is done by way of Apple's
 to provide client multiplexing, signature validation of connecting clients and
 forced connection establishment. This is called SNTXPCConnection.
 
-##### Who starts who?
+## Who starts who?
 
-The santad and Santa (GUI) processes are both started and kept alive by launchd
-as a LaunchDaemon and a LaunchAgent, respectively. This means santad runs as
-root and Santa (GUI) runs as the console user.
+The primary Santa daemon, `santad` is a System Extension. It gets submitted as a
+`launchd(8)` job by `endpointsecurityd(8)` via `sysextd(8)`. This happens at
+system startup or when Santa is installed/upgraded.
 
-There can be multiple Santa (GUI) processes running, one per user logged into
-the GUI (assuming fast-user switching is enabled). While multiple processes
+The `Santa` GUI component runs as a launch agent when a user logs into the
+system. There can be multiple Santa (GUI) processes running, one per user logged
+into the GUI (assuming fast-user switching is enabled). While multiple processes
 might be running, only the one for the user currently logged-in will be
-connected to santad and receiving notifications.
+connected to `santad` and receiving notifications.
 
-When using a sync server, the santactl process is started by santad. Before the
-new process starts, all privileges are dropped. santactl runs as _nobody_.
+Both `santasyncservice` and `santabundleservice` run as launch daemons. If a
+sync server is configured, `santasyncservice` gets launched when `santad` starts
+up and establishes an XPC connection. The `santabundleservice` starts up when a
+binary is blocked and the `Santa` GUI establishes an XPC connection. Note, the
+`santabundleservice` also communicates back with `santad`.
 
-The santabs process is started by launchd via an XPC service connection from
-santad. XPC services inherit their initiator's privileges meaning santabs runs
-as root, which is necessary to ensure it has permission to read all files.
+The `santametricservice` also runs as a launch daemon and starts when the system
+comes up. `santad` opens and maintains an XPC connection it.
```

Comment:
Why is this one separate from syncservice and bundleservice? If we're going to group them I'd argue sync and metric service should be grouped because they're both standard daemon's that run constantly, while bundleservice is on-demand.

Reply:
Sync service isn't `RunAtLoad` or `KeepAlive` which was why I grouped them the way I have them. But I personally don't see a particular benefit of grouping them one way or the other, happy to combine them.